src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractViolationReporterTest.java

```diff
@@ -133,6 +134,18 @@ public void testCustomMessageWithParametersNegative() throws Exception {
         }
     }
 
+    /*
+      Due to low level configuration setup of AbstractViolationReporter, conventional
```

Comment:
Explanation here is not clear, please elaborate "low level configuration" and what conditions caused us to use pure unit test.

Reply:
@nrmancuso 
This explanation was suggested by @romani.
I mean that message is a bit unclear to me too. I noticed that the other methods in the test file are also pure JUnit tests and no tests are written using the BDD model (using ``verifyWithInlineConfigParser()`` etc). Can you suggest any message?
My suggestion:
``AbstractViolationReporter has not been covered under the BDD model. Conventional input validation cannot be done here hence, pure JUnit testing has been done for the time being.``

Reply:
@nrmancuso @romani is th message I have suggested good enough. Or should I write something else.

Reply:
@romani please tell what message to write here as there is no toString() method in this as there is in the other PR i have made.

Reply:
It sounds like we can create Input test that does set severity as Ignore or some other value.

Junit is not appropriate here

Reply:
Ok i will try and push changes

Reply:
@romani I have tried to not use assertions here but, please check the code I have pushed. 

It wouldn't let me write the test this way as ``SetSeverity`` class that I have written in ``AbstractViolationReporterTest``  is not allowed as a child in checker. 

```shell
[ERROR] Errors: 
[ERROR]   AbstractViolationReporterTest.testSetSeverity:89->AbstractModuleTestSupport.verify:280->AbstractModuleTestSupport.createChecker:105->AbstractMo
duleTestSupport.configureChecker:129 ┬╗ Checkstyle com.puppycrawl.tools.checkstyle.api.AbstractViolationReporterTest$SetSeverity is not allowed as a child in Checker
[INFO]
[ERROR] Tests run: 3976, Failures: 0, Errors: 1, Skipped: 1
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  04:39 min
[INFO] Finished at: 2023-04-05T11:54:43+05:30
[INFO] ------------------------------------------------------------------------
```

Reply:
@shubh220922 I think we have some confusion around severity, please take a look at https://checkstyle.sourceforge.io/config.html#Severity. Can we use some implementation of AbstractViolationReporter with different severity level to kill mutation? It would be best to use one that already has a target test in `pitest-api`